shakytweaks/src/main/java/com/mindsea/shakytweaks/ShakyTweaks.kt

```diff
@@ -124,5 +145,30 @@ object ShakyTweaks {
         return (Date().time - pastKeyPressMS) <= KEY_PRESS_THRESHOLD
     }
 
+    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+        when (event) {
```

Comment:
This method will only be ever called _after_ `lifecycle.addObserver` is executed, so I think it is safe to remove the `isInitialized` checks.

Also, IMO having `lateinit` vars so that they aren't optional, but then having to use `::var.isInitalized` does't make sense, with this we are basically back to Java with no enforced null checks. If there is any chance of the var to be optional, I prefer to leave it as optional and force the developer to make null checks. On top of that, you can make some helper methods that throw an exception if the var "should" not be optional at that moment... just like `requireContext()` do.

Reply:
@gianei  I leave them optional because the method below is public. If the user calls without call the init method first, the app crashes.

```
override fun onStateChanged
```

Reply:
🤔 
If it is optional, then I would remove the `lateinit`, and throw errors with custom messages explaining that this method should not be called directly.

But, you can do better:
Instead of making `ShakyTweaks` implement `LifecycleEventObserver`, try moving it to a inner class or a separate class. I think that by moving to a separate class, you could mark it as `private`, `protected`, or `internal`. Then no-one outside this lib would possibly call this. 
Give it a try 😬

Reply:
Amazing @gianei . I did what you said, it looks better now 🤗️
Thanks